src/AppInstallerCommonCore/Public/winget/ManifestValidation.h

```diff
@@ -51,6 +51,9 @@ namespace AppInstaller::Manifest
         const char* const ExceededAppsAndFeaturesEntryLimit = "Only zero or one entry for Apps and Features may be specified for InstallerType portable.";
         const char* const ExceededCommandsLimit = "Only zero or one value for Commands may be specified for InstallerType portable.";
         const char* const ScopeNotSupported = "Scope is not supported for InstallerType portable.";
+        const char* const ApproximateVersionNotAllowed = "Approximate version not allowed.";
+        const char* const ArpVersionOverlapWithIndex = "DisplayVersion declared in the manifest has overlap with existing DisplayVersion range in the index. Existing DisplayVersion range in index: ";

```

Comment:
This got me thinking - Are version ranges architecture specific, and/or should they be? I could see a scenario where a publisher uses the following - 
```yaml
PackageVersion: 1.0-Alpha # This is their marketing version
Installers: 
  - Architecture: x86
    AppsAndFeaturesEntries:
      - DisplayVersion: 1.0.4.0
- Architecture: x64
  AppsAndFeaturesEntries:
    - DisplayVersion: 1.0.5.0
```
According to the spec, this would result in an Version Range of [1.0.4.0, 1.0.5.0]

Suppose they release a new update - 
```yaml
PackageVersion: 1.1-Alpha # This is their marketing version
Installers: 
  - Architecture: x86
    AppsAndFeaturesEntries:
      - DisplayVersion: 1.0.4.1
- Architecture: x64
  AppsAndFeaturesEntries:
    - DisplayVersion: 1.0.5.1
```
According to the logic, wouldn't 1.0.4.1 overlap with [1.0.4.0, 1.0.5.0]. since 1.0.4.0 =< 1.0.4.1 =< 1.0.5.0, thereby triggering a false overlap warning?

Reply:
Correct, this will be treated as unsupported case according to the spec previously written. Since there's no restriction on what is written to DisplayVersion entry, an installer could also write different values for different locale. The above example can be extended to scope, installertype, etc. Essentially this will end up being each installer is in its own set.